### PR TITLE
Add CORS headers to http responses

### LINKED ISSUE #638: Add CORS headers to http responses
At the moment, the daemon adds CORS headers only to [pre-flight responses](https://github.com/tdex-network/tdex-daemon/blob/master/internal/interfaces/grpc/http_utils.go#L286-L288) while it should do the same also for [GET responses](https://github.com/tdex-network/tdex-daemon/blob/master/internal/interfaces/grpc/http_utils.go#L291-L293) and for [grpc-gateway endpoints](https://github.com/tdex-network/tdex-daemon/blob/master/internal/interfaces/grpc/http_utils.go#L303-L305).

### PR DESCRIPTION
**Pull request by sekulicd**
This adds CORS headers to HTTP responses.

This closes #638 

@tiero @altafan please review.

### CODE REVIEW

**Comment by tiero** [comment]
what about grpc-web requests? Are them already responding with cors headers?

**Comment by tiero** [comment]
Also what about max-age @altafan ? could that be important to be set with Tor deployment that can take more than 5 second to respond? (I think here is just to avoid to much preflight request in case of latency)

**Comment by altafan** [comment]
> what about grpc-web requests? Are them already responding with cors headers?

`grpcweb` pjg already takes care of [cors headers](https://github.com/tdex-network/tdex-daemon/blob/master/internal/interfaces/grpc/http_utils.go#L279-L283)

**Review by altafan — APPROVED** [review]

File: internal/interfaces/grpc/http_utils.go [hunk 1]
```diff
@@ -302,6 +304,8 @@ func newGRPCWrappedServer(
 
 		if grpcGateway != nil {
 			if isHttpRequest(req) {
+				w.Header().Set("Access-control-Allow-Origin", "*")
+				w.Header().Set("Access-control-Allow-Headers", "*")
```
Comment (tiero) [inline internal/interfaces/grpc/http_utils.go:308]:
```suggestion
				w.Header().Set("Access-Control-Allow-Origin", "*")
				w.Header().Set("Access-Control-Allow-Headers", "*")
```

**Comment by tiero** [comment]
what about `Access-Control-Allow-Methods`

**Comment by altafan** [comment]
it's actually missing.. we should allow only GET, POST?

File: internal/interfaces/grpc/http_utils.go (hunk 1 above)
Comment (sekulicd) [reply internal/interfaces/grpc/http_utils.go]: Done

File: internal/interfaces/grpc/http_utils.go [hunk 2]
```diff
@@ -290,6 +290,8 @@ func newGRPCWrappedServer(
 		}
 		if isGetRequest(req) {
 			if handler, ok := httpHandlers[req.URL.Path]; ok {
+				w.Header().Set("Access-control-Allow-Origin", "*")
+				w.Header().Set("Access-control-Allow-Headers", "*")
```
[the remark this answers is not in the document]
Comment (sekulicd) [reply internal/interfaces/grpc/http_utils.go]: Done

**Comment by sekulicd** [comment]
> what about `Access-Control-Allow-Methods`

Added

File: internal/interfaces/grpc/http_utils.go [hunk 3]
```diff
@@ -284,12 +284,16 @@ func newGRPCWrappedServer(
...
+			w.Header().Set("Access-Control-Allow-Origin", "*")
+			w.Header().Set("Access-Control-Allow-Headers", "*")
+			w.Header().Set("Access-Control-Allow-Methods", "*")
 			return
 		}
 		if isGetRequest(req) {
 			if handler, ok := httpHandlers[req.URL.Path]; ok {
+				w.Header().Set("Access-Control-Allow-Origin", "*")
+				w.Header().Set("Access-Control-Allow-Headers", "*")
+				w.Header().Set("Access-Control-Allow-Methods", "*")
```
Comment (altafan) [inline internal/interfaces/grpc/http_utils.go:296]: @sekulicd must be only GET, POST, OPTIONS, not all methods

**Review by tiero — APPROVED** [review]

**Pull request merged**